client/dashboard/me/billing-purchases/payment-methods/use-memo-compare.ts

```diff
@@ -0,0 +1,22 @@
+import { useEffect, useRef } from 'react';
+
+/**
+ * Custom hook that memoizes a value based on a comparison function
+ * @param value The value to memoize
+ * @param compare Function to compare previous and next values
+ * @returns The memoized value
+ */
+export function useMemoCompare< T >( value: T, compare: ( prev: T, next: T ) => boolean ): T {
+	const previousRef = useRef< T >( value );
+	const previous = previousRef.current;
+
+	const isEqual = compare( previous, value );
```

Comment:
Should this check to see if the `value !== previousRef.current` before running `compare`?